### PR TITLE
feature: delete prev rt rows when indexing

### PR DESCRIPTION
**Pull request by TamarinEA**
https://jira.railsc.ru/browse/GOODS-2250

### CODE REVIEW

File: spec/sphinx/integration/transmitter_spec.rb [hunk 1]
```diff
@@ -36,6 +36,26 @@
...
+            {'sphinx_internal_id' => 1, 'region_id' => '123', 'has_region' => 't'}
+           ])
+          expect(client).to receive(:write).with(
+            'REPLACE INTO model_with_rt_rt0 (`sphinx_internal_id`, `region_id`, `has_region`, `rubrics`)' \
+              ' VALUES (1, 123, 1, ())'
+          )
+          expect(client).to receive(:write).
+            with("UPDATE model_with_rt_core SET sphinx_deleted = 1 WHERE " \
+               "`id` IN (#{record.sphinx_document_id}) AND `sphinx_deleted` = 0")
+          expect(client).to receive(:write).with("DELETE FROM model_with_rt_rt1 WHERE id = #{record.sphinx_document_id}")
```
Comment (apress-hound) [inline spec/sphinx/integration/transmitter_spec.rb]: Metrics/LineLength: Line is too long. [121/120]

**Comment by TamarinEA** [comment]
@isqad @c1n1c @imhouston

**Comment by TamarinEA** [comment]
@isqad @c1n1c

File: lib/sphinx/integration/extensions/thinking_sphinx/index.rb [hunk 2]
```diff
@@ -168,8 +171,12 @@ def indexing?
     mutex(:indexing).locked?
   end
 
-  def indexing
-    mutex(:indexing).with_lock { yield }
+  def indexing(need_lock = true)
```
Comment (isqad) [inline lib/sphinx/integration/extensions/thinking_sphinx/index.rb]: может `def indexing(need_lock: true)`

File: lib/sphinx/integration/statements/distributed.rb [hunk 3]
```diff
@@ -9,27 +9,23 @@ def initialize(index)
         end
 
         def update(data, matching: nil, where: {})
-          index_names do |index_name|
-            sql = ::Sphinx::Integration::Extensions::Riddle::Query::Update.
-              new(index_name, data, where.merge!(sphinx_deleted: 0), prepare_matching(matching)).
-              to_sql
+          sql = ::Sphinx::Integration::Extensions::Riddle::Query::Update.
+            new(index_name, data, where.merge!(sphinx_deleted: 0), prepare_matching(matching)).
```
Comment (isqad) [inline lib/sphinx/integration/statements/distributed.rb]: обычно hound ругался на такое
```
where[:sphinx_deleted] = 0

sql = ::Sphinx::Integration::Extensions::Riddle::Query::Update.
            new(index_name, data, where, prepare_matching(matching)).to_sql
```

**Pull request merged**